src-built-in/components/singleInputDialog/src/app.jsx

```diff
@@ -6,140 +6,15 @@ import React from "react";
-	 *
-	 * @param {any} err
-	 * @param {any} response
-	 * @memberof SingleInputDialog
-	 */
-	onShowRequested(err, response) {
-		let data = response.data;
-		this.setState({
-			title:data.title || "Title",
-			inputValue: null,
-			hideModalOnClose: typeof data.hideModalOnClose === "undefined" ? true : data.hideModalOnClose,
-			inputLabel: data.inputLabel,
-			affirmativeResponseLabel: data.affirmativeResponseLabel || "OK",
-			cancelResponseLabel: data.cancelResponseLabel || "Cancel",
-			showCancelButton: typeof data.showCancelButton === "undefined" ? false : data.showCancelButton,
-			renderInput: true,
-		}, this.fitAndShow);
-	}
-	/**
-	 * Fits the contents of the DOM to the window, then calls `showDialog`, which positions the dialog on the proper monitor and toggles the visibility of the window.
-	 *
-	 * @memberof SingleInputDialog
-	 */
-	fitAndShow() {
-		FSBL.Clients.WindowClient.fitToDOM(null, function () {
-			FSBL.Clients.DialogManager.showDialog();
-		});
-	}
-	/**
-	 * Sends user input to the opener.
-	 *
-	 * @param {any} response
-	 * @memberof SingleInputDialog
-	 */
-	sendResponse(response) {
-		FSBL.Clients.DialogManager.respondToOpener({
-			value: this.state.inputValue,
-			choice: response,
-			hideModalOnClose: this.state.hideModalOnClose
-		});
-
-		this.setState({
-			inputValue: null,
-			renderInput: false
-		});
-		Array.from(document.querySelectorAll("input")).forEach((el) => el.value = "");
-	}
-	/**
-	 * onChange handler. When the user types, we keep the value up to date in the dialog's state.
-	 *
-	 * @param {any} e
-	 * @memberof SingleInputDialog
-	 */
-	setInputValue(e) {
-		this.setState({
-			inputValue: e.target.value
-		});
-	}
-
-	render() {
-		var self = this;
-		return (<FinsembleDialog
-			userInputTimeout={10000}
-			behaviorOnResponse="hide"
-			onShowRequested={this.onShowRequested}
-			isModal={true}>
-			<div className="dialog-title">{this.state.title}</div>
-			{this.state.renderInput &&
-				<FinsembleDialogTextInput onInputChange={this.setInputValue} placeholder="New Workspace" autoFocus={true} />
-			}
-			<div className="button-wrapper">
-				{
-					this.state.showCancelButton &&
-					<FinsembleDialogButton className="fsbl-button-neutral" onClick={() => {
-						this.sendResponse("cancel");
-					}}>
-						{this.state.cancelResponseLabel}
-					</FinsembleDialogButton>
-				}
-				<FinsembleDialogButton className="fsbl-button-affirmative" onClick={() => { this.sendResponse("affirmative"); }}>
-					{this.state.affirmativeResponseLabel}
-				</FinsembleDialogButton>
-			</div>
-		</FinsembleDialog>);
-	}
-}
+import { SingleInputDialog } from "@chartiq/finsemble-ui/lib/components/singleInputDialog/singleInputDialog";
 
 //render component when FSBL is ready.
-if (window.FSBL && FSBL.addEventListener) { FSBL.addEventListener("onReady", FSBLReady); } else { window.addEventListener("FSBLReady", FSBLReady) }
+
 function FSBLReady() {
```

Comment:
We can hide this FSBLReady function inside finsemble UI and only renders it when FSBL is ready

Reply:
Done.